app/lib/actions/middleware/watch_delegated_proxy_sub_tasks.rb

```diff
@@ -0,0 +1,72 @@
+module Actions
+  module Middleware
+    class WatchDelegatedProxySubTasks < ::Dynflow::Middleware
+      class CheckOnProxyActions; end
+      # Poll the proxy every 10 minutes
+      POLL_INTERVAL = 600
+      BATCH_SIZE = 1000
+
+      def run(event = nil)
+        if event.nil?
+          set_clock
+        elsif event == CheckOnProxyActions
+          check_triggered
```

Comment:
In case `check_triggered` raises expection (such as in case the proxy itself is down, which is something I've tested), the next `CheckOnProxyActions` in not triggered, and we would not check for the presence once the proxy is available again. We should handle this case

Reply:
more importantly, the unhandled exception would actually lead the whole task to fail

Reply:
I thought that would be handled here https://github.com/theforeman/foreman-tasks/pull/298/files#diff-1316c50efc730290525bdcb5bf816560R66 , but maybe another exception is raised

Reply:
Please give it a try on your setup: there is still some chance there would be something weird on my setup: The error was one of `Errno::ECONNREFUSED, Errno::EHOSTUNREACH, Errno::ENETUNREACH` and despite the fact that it seems it should inherit from `StandardError`, it's not being caught it seems, but I welcome additional try of someone else.